include/cantera/thermo/Phase.h

```diff
@@ -471,6 +471,23 @@ class Phase
     //! units = kg / kmol
     const vector_fp& molecularWeights() const;
 
+    //! Charge of species \c k.
+    //!     @param k index of species \c k
+    //!     @returns the charge of species \c k.
+    doublereal speciesCharge(size_t k) const;
+
+    //! Copy the vector of species charges into vector charges.
+    //!     @param charges Output vector of species charges (-)
+    void getSpeciesCharges(vector_fp& charges) const;
+
+    //! Copy the vector of species charges into array weights.
+    //!     @param weights Output array of species charges (-)
+    void getSpeciesCharges(doublereal* charges) const;
```

Comment:
Please use `double` rather than `doublereal` in any new code (but don't change the use of the `doublereal` macro on any lines that would otherwise be unmodified).

Presumably `weights` in the summary line should be `charges`?

Is the notation `(-)` meant to suggest that this is dimensionless? I think it would be more correct to say that this is in units of the elementary charge.